CefSharp.Core/Internals/ClientAdapter.cpp

```diff
@@ -323,15 +350,36 @@ namespace CefSharp
 
         bool ClientAdapter::OnBeforeBrowse(CefRefPtr<CefBrowser> browser, CefRefPtr<CefFrame> frame, CefRefPtr<CefRequest> request, bool isRedirect)
         {
-            IRequestHandler^ handler = _browserControl->RequestHandler;
-            if (handler == nullptr)
+            if (browser->IsPopup())
             {
-                return false;
+                auto popupHandler = _browserControl->PopupHandler;
+                if (popupHandler != nullptr)
+                {
+                    auto browserId = browser->GetIdentifier();
+                    IBrowser^ entry;
+                    if (_popupBrowsers->TryGetValue(browserId, entry))
+                    {
+                        auto frameWrapper = gcnew CefFrameWrapper(frame, _browserAdapter);
+                        CefRequestWrapper^ wrapper = gcnew CefRequestWrapper(request);
+                        return popupHandler->OnBeforeBrowse(_browserControl, entry, wrapper, isRedirect, gcnew CefFrameWrapper(frame, _browserAdapter));
```

Comment:
Is it practical to cache the `CefFrameWrapper` instance? What's the trigger to dispose of an instance? 

Obviously the simplest approach from our end is to dispose straight after callback. How limiting would that be? We've got the same problem with `GetFocusedFrame` and others. Then we have even less control. 

Does `OnLoadingStateChange` (or similar) give us a clear boundary to dispose of old instances?